webapp/components/survey/SurveyForm/SurveyForm.js

```diff
@@ -75,6 +77,13 @@ const SurveyForm = (props) => {
     }
   }, [])
 
+  useHistoryListen((location) => {
+    // user enters in records the form is reset
+    if (matchPath(location.pathname, { path: appModuleUri(dataModules.records) })) {
+      dispatch(SurveyFormActions.resetForm())
```

Comment:
can you check if dispatching this action only on the FormDesigner unmount might work?